### PR TITLE
[MM-24507] - Pin comment box on bottom of RHS thread

### PR DESCRIPTION
**Pull request by nevyangelova**
#### Summary
Pin comment box on bottom of RHS thread. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24507

### CODE REVIEW

**Comment by esethna** [comment]
@nevyangelova can you test this on chrome Win 10? RHS stops scrolling after the comment thread gets large enough to push the comment box to the bottom of the screen

image

**Comment by nevyangelova** [comment]
Sorry @esethna i have somehow missed the comment. It's working for me on all devices now :)

**Comment by esethna** [comment]
Thanks, adding @jgilliam17 for testing

**Review by calebroseland — APPROVED** [review]
LGTM!

**Comment by jgilliam17** [comment]
Thank you @nevyangelova 
Tested on Desktop app, Chrome, Firefox and Edge on macOS Catalina,  - works as expected.
Tested on Desktop app, Edge on Windows 10 - works as expected.
Tested on Desktop app and Firefox on Ubuntu - works as expected.
1. Safari on macOS Catalina has an issue, see gif. RHS fails to scroll up when adding additional posts. 
Sep-16-2020 21-35-46

2. May not be an issue - Observed: RHS scrolls to the bottom as soon as you add text, not when posting. Not sure if this is intended, differs from main channel. @esethna any thoughts?
i. Open long thread on the RHS
ii. Scroll up in the thread, close to top
iii. Enter at least one character into RHS box
Observed: RHS scrolls to the bottom as soon as you start typing.
Sep-16-2020 22-14-58

**Comment by nevyangelova** [comment]
@jgilliam17 seems like another safari weirdness but I can't repro :( any additional steps?

**Comment by esethna** [comment]
@jgilliam17 @nevyangelova re #2, that looks like a regression

**Comment by jgilliam17** [comment]
Nevy,
For 1. I don't have any additional steps. For me in Safari the RHS thread is stuck on top and does not scroll at all to reveal new posts.
For 2. Just wanted to add that I am seeing this on all devices (other than Safari)

**Comment by nevyangelova** [comment]
Good to know I will add a scrollBottom functionality if the component doesn't already have a prop for that! Thanks both

**Comment by amyblais** [comment]
Pushing this from v5.28 as it will probably not make it to today's feature complete. Let me know any concerns,

**Review by esethna — APPROVED** [review]

**Comment by esethna** [comment]
@jgilliam17 please help test on various browsers and desktop :)

File: components/rhs_thread/rhs_thread.tsx [hunk 1]
```diff
@@ -385,6 +386,8 @@ export default class RhsThread extends React.Component<Props, State> {
                     autoHide={true}
                     autoHideTimeout={500}
                     autoHideDuration={500}
+                    autoHeight={true}
+                    autoHeightMax={'calc(100vh - 190px - 56px - 62px)'}
```
Comment (hmhealey) [inline components/rhs_thread/rhs_thread.tsx:390]: Could you put a comment somewhere here to explain where these numbers come from? I assume it's something like the channel header, comment box, and rhs header

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: components/rhs_thread/rhs_thread.tsx (hunk 1 above)
Comment (nevyangelova) [reply components/rhs_thread/rhs_thread.tsx:390]: Indeed :D should have done that 👍

**Review by hmhealey — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @nevyangelova 
Tested, looks good to merge. 🎉
- Verified text box is pinned to the bottom of the RHS on long threads. RHS thread scrolls to the bottom when new post is added. Tested on various browsers and desktop apps on macOS, Windows and Ubuntu

**Pull request merged**